tests/Types/TypeRegistryTest.php

```diff
@@ -99,6 +99,16 @@ public function testRegisterWithAlreadyRegisteredInstance(): void
         $this->registry->register('other', $newType);
     }
 
+    public function testConstructorWithDuplicateInstance(): void
+    {
+        $newType = new TextType();
+
+        new TypeRegistry(['a' => $newType]);
+
+        $this->expectException(Exception::class);
+        new TypeRegistry(['a' => $newType, 'b' => $newType]);
+    }
```

Comment:
I believe the first constructor call should already be covered by other tests. If that's not the case, please move it into a separate test and run assertions on the constructed instance.

Reply:
It is definitely covered, I added `new TypeRegistry(['a' => $newType]);` here to assert that the constructor pass otherwise (with single test) - you want me to remove it?

Reply:
yes please

Reply:
addressed